virttest/qemu_vm.py

```diff
@@ -1321,6 +1321,24 @@ def add_disable_modern(devices, dev, dev_type):
                 value = params.get("disable_modern", "on")
                 dev.set_param("disable-modern", value)
 
+        def add_iommu_platform(devices, dev, dev_type):
```

Comment:
We may need to refactor these four functions as they are almost same.

Reply:
updated,  use a common function instead. Could you help review again? thanks a lot!